config/cloud.cfg.tmpl

```diff
@@ -191,34 +192,155 @@ cloud_final_modules:
    distro: {{ variant }}
-{% elif variant in ["dragonfly"] %}
+{% elif variant == "dragonfly" %}
    distro: dragonflybsd
 {% else %}
    # Unknown/fallback distro.
    distro: ubuntu
 {% endif %}
-{% if variant in ["ubuntu", "unknown", "debian"] %}
+{% if variant in ["alpine", "amazon", "arch", "debian", "fedora", "gentoo",
+                  "OpenCloudOS", "openEuler", "openmandriva", "suse",
+                  "TencentOS", "ubuntu", "unknown"] or is_rhel %}
    # Default user name + that default users groups (if added/used)
+{% endif %}
    default_user:
+{% if variant == "amazon" %}
+     name: ec2-user
+{% elif variant in ["centos", "rhel"] %}
+     name: cloud-user
+{% elif variant == "openmandriva" %}
+     name: omv
+{% elif variant == "unknown" %}
      name: ubuntu
+{% else %}
+     name: {{ variant }}
+{% endif %}
+{% if variant in ["alpine", "amazon", "arch", "debian", "fedora", "gentoo",
+                  "mariner", "OpenCloudOS", "openEuler", "openmandriva",
+                  "photon", "suse", "TencentOS", "ubuntu", "unknown"]
+                 or is_bsd or is_rhel %}
      lock_passwd: True
+{% endif %}
+{% if variant == "amazon" %}
+     gecos: EC2 Default User
+{% elif variant in ["centos", "rhel"] %}
+     gecos: Cloud User
+{% elif variant == "debian" %}
+     gecos: Debian
+{% elif variant == "dragonfly" %}
+     gecos: DragonFly
+{% elif variant == "freebsd" %}
+     gecos: FreeBSD
+{% elif variant == "mariner" %}
+     gecos: MarinerOS
+{% elif variant == "netbsd" %}
+     gecos: NetBSD
+{% elif variant == "openbsd" %}
+     gecos: OpenBSD
+{% elif variant == "openmandriva" %}
+     gecos: OpenMandriva admin
+{% elif variant == "photon" %}
+     gecos: PhotonOS
+{% elif variant in ["ubuntu", "unknown"] %}
      gecos: Ubuntu
+{% else %}
+     gecos: {{ variant }} Cloud User
+{% endif %}
+{% if variant == "alpine" %}
+     groups: [adm, wheel]
+{% elif variant == "arch" %}
+     groups: [wheel, users]
+{% elif variant in ["debian", "ubuntu", "unknown"] %}
      groups: [adm, cdrom, dip, lxd, sudo]
+{% elif variant == "gentoo" %}
+     groups: [users, wheel]
+     primary_group: users
+     no_user_group: true
+{% elif variant in ["mariner", "photon"] or is_bsd %}
+     groups: [wheel]
+{% elif variant == "openmandriva" %}
+     groups: [wheel, users, systemd-journal]
+{% elif is_rhel %}
+     groups: [adm, systemd-journal]
+{% elif variant == "suse" %}
+     groups: [cdrom, users]
+{% elif not is_bsd %}
+     groups: [wheel, adm, systemd-journal]
+{% endif %}
      sudo: ["ALL=(ALL) NOPASSWD:ALL"]
+{% if variant == "alpine" %}
+     shell: /bin/ash
+{% elif variant in ["amazon", "arch", "debian", "fedora", "gentoo", "mariner",
+                    "OpenCloudOS", "openEuler", "openmandriva", "photon",
+                    "suse", "TencentOS", "ubuntu", "unknown"] or is_rhel %}
+     shell: /bin/bash
+{% elif variant in ["dragonfly", "netbsd"] %}
+     shell: /bin/sh
+{% elif variant == "freebsd" %}
+     shell: /bin/tcsh
```

Comment:
didn't realise this was our default shell.
probably because i usually change it to fish…
the "new" default shell for quite some time is "/bin/sh"

(dunno what it is in dragonfly, will try to investigate)

Reply:
let's fold all the BSDs except OpenBSD into one case

Reply:
the gecos/groups/shells/usernames stuff has been reworked in general.

Is there some other BSD stuff you think should be changed?